File: data_structures/Queue/Queue.js



class Queue {
  constructor() {
    this.queue = [];
  }

  add(val) {
    this.queue.unshift(val);
  }

  remove() {
    if (this.queue.length !== 0) {
      this.queue.pop();
    } else {
      console.log("Queue is empty !")
    }
  }

  peek() { // returns the next element in the queue
    return this.queue[this.queue.length - 1];
  }

  getData() {
    return this.queue;
  }
}

const newQueue = new Queue()

console.log("Enqueueing !")
console.log(newQueue.getData())
newQueue.add(23)
console.log(newQueue.getData())
newQueue.add(2)
console.log(newQueue.getData())
newQueue.add(45)
console.log(newQueue.getData())
newQueue.add(7)
console.log(newQueue.getData())
newQueue.add(0)
console.log(newQueue.getData())
newQueue.add(5)
console.log(newQueue.getData())

console.log("Dequeueing data !")
console.log(newQueue.getData())
newQueue.remove()
console.log(newQueue.getData())
newQueue.remove()
console.log(newQueue.getData())
newQueue.remove()
console.log(newQueue.getData())
newQueue.remove()
console.log(newQueue.getData())
newQueue.remove()
console.log(newQueue.getData())
newQueue.remove()
console.log(newQueue.getData())
newQueue.remove()



function weave (queue1, queue2) {

  let q = new Queue()

  while (queue1.peek() || queue2.peek()) {
    if(queue1.peek()){
      q.add(queue1.remove())
    }
    if(queue2.peek()){
      q.add(queue2.remove())
    }
  }

  return q.getData()
}

File: data_structures/Queue/Queue.py


class Queue:
    def __init__(self):
        self.queue = []

    def add(self, val):
        self.queue.append(val)

    def remove(self):
        if len(self.queue) == 0:
            print("Queue is Empty !")
        else:
            self.queue.pop(0)

    def getData(self):
        print(self.queue)


newQueue = Queue()

print("Enqueueing !")
newQueue.getData()
newQueue.add(4)
newQueue.getData()
newQueue.add(23)
newQueue.getData()
newQueue.add(7)
newQueue.getData()
newQueue.add(0)
newQueue.getData()
newQueue.add(11)
newQueue.getData()

print("Dequeueing !")
newQueue.getData()
newQueue.remove()
newQueue.getData()
newQueue.remove()
newQueue.getData()
newQueue.remove()
newQueue.getData()
newQueue.remove()
newQueue.getData()
newQueue.remove()
newQueue.getData()
newQueue.remove()



def weave(queue1, queue2):

    q = Queue()

    while queue1.peek() or queue2.peek():
        if queue1.peek():
            q.add(quque1.remove())
        if queue2.peek():
            q.add(queue2.remove())

    return q.getData()








